base/reflection.jl

```diff
@@ -754,6 +754,8 @@ function ismutationfree(@nospecialize(t))
         return datatype_ismutationfree(t)
     elseif isa(t, Union)
         return ismutationfree(t.a) && ismutationfree(t.b)
+    elseif isa(t, Core.TypeofVararg)
+        return true
```

Comment:
This doesn't seem quite true, since both the `T` and `N` field can point to mutable types (i.e. DataType)

Reply:
Emm... I must say I get confused by the doc string "no mutable memory is reachable from this type (either in the type itself)"
We have
```julia
julia> Base.ismutationfree(DataType)
true

julia> Base.ismutationfree(Type{Int})
true
```
though.

Reply:
Oh, I guess we must special-case DataType to be slightly incorrect here? So maybe this is fine, since it is consistent with DataType already anyways
```
julia> Base.ismutationfree(DataType)
true

julia> Base.ismutationfree(Core.TypeName)
false
```